Mergin/plugin.py

```diff
@@ -575,71 +625,116 @@ def __init__(self, parent, grp_name, grp_filter, icon, order, project_manager):
+
+    def fetch_projects(self, page=1, per_page=PROJS_PER_PAGE):
+        """Get paginated projects list from Mergin service and an error message list."""
+        if self.project_manager is None:
             error_item = QgsErrorItem(self, "Failed to login please check the configuration", "/Mergin/error")
             sip.transferto(error_item, self)
             return [error_item]
         try:
-            projects = mc.projects_list(flag=self.filter)
+            resp = self.project_manager.mc.paginated_projects_list(
+                flag=self.filter, page=page, per_page=per_page, order_by="namespace")
+            self.projects += resp["projects"]
+            self.total_projects_count = int(resp["count"]) if is_number(resp["count"]) else 0
         except URLError:
             error_item = QgsErrorItem(self, "Failed to get projects from server", "/Mergin/error")
             sip.transferto(error_item, self)
             return [error_item]
         except Exception as err:
             error_item = QgsErrorItem(self, "Error: {}".format(str(err)), "/Mergin/error")
             sip.transferto(error_item, self)
-            return [error_item]
+            return None, [error_item]
+        return None
 
+    def createChildren(self):
+        if not self.projects:
+            error = self.fetch_projects()
+            if error:
+                return error
         items = []
-        for project in projects:
-            item = MerginProjectItem(self, project, mc, self.project_manager)
+        for project in self.projects:
+            item = MerginProjectItem(self, project, self.project_manager)
             item.setState(QgsDataItem.Populated)  # make it non-expandable
             sip.transferto(item, self)
             items.append(item)
+        self.set_fetch_more_item()
+        if self.fetch_more_item is not None:
+            items.append(self.fetch_more_item)
         return items
 
+    def set_fetch_more_item(self):
+        """Check if there are more projects to be fetched from Mergin service and set the fetch-more item."""
+        if self.fetch_more_item is not None:
+            try:
+                self.removeChildItem(self.fetch_more_item)
+            except RuntimeError:
+                pass
+            self.fetch_more_item = None
+        fetched_count = self.rowCount() + len(self.projects)
+        if fetched_count < self.total_projects_count:
+            self.fetch_more_item = FetchMoreItem(self)
+            self.fetch_more_item.setState(QgsDataItem.Populated)
+            sip.transferto(self.fetch_more_item, self)
+        group_name = f"{self.group_name} ({fetched_count}/{self.total_projects_count})"
+        self.setName(group_name)
+
+    def fetch_more(self):
+        """Fetch another page of projects and add them to the group item."""
+        if self.total_projects_count - self.rowCount() <= 0:
+            QMessageBox.information(None, "Fetch Mergin Projects", "All projects already listed.")
```

Comment:
Can this ever happen? Do we allow people to fetch more if there's nothing else to fetch?

Reply:
Nope, this is a leftover from an earlier stage when the "Fetch more..." item was not removed from the list when all projects were already fetched.
Fixed.